api/v1alpha1/database_types.go

```diff
@@ -110,6 +110,57 @@ type Instance struct {
 	Type string `json:"type"`
 	// +optional
 	Profiles Profiles `json:"profiles"`
+	// +optional
+	// +kubebuilder:default:="{emptytminfo}"
+	// Information related to time machine that is to be associated with this database
+	TMInfo TimeMachineInfo `json:"timeMachine"`
+}
+
+// Time Machine details
+type TimeMachineInfo struct {
```

Comment:
should we move this to a diff file under the same package?

Reply:
Yes we can! But I kept it under `database_types.go` because this is closely related to the database spec. If I move this out to something like TimeMachine/TimeMachineInfo.go and sometime down the line when we introduce the TimeMachine CR, then that will create something like `timemachine_types.go`, this might create some confusion which is what I was trying to avoid. Anyway, let me know how you feel about this.

Reply:
talked offline - we will rename this struct and keep it at the same location